libbpfgo.go

```diff
@@ -428,20 +417,20 @@ func (m *Module) Close() {
 }
 
 func (m *Module) BPFLoadObject() error {
-	ret := C.bpf_object__load(m.obj)
-	if ret != 0 {
-		return fmt.Errorf("failed to load BPF object")
+	cErr := C.bpf_object__load(m.obj)
+	if cErr != 0 {
+		return fmt.Errorf("failed to load BPF object", syscall.Errno(int(-cErr)))
```

Comment:
Is this right ? Follow my thoughts:

You call bpf_object__load(NULL)... then you will face:

```C
if (!attr)
        return libbpf_err(-EINVAL);
```

and `libbpf_err(-EINVAL):`:

```C
static inline int libbpf_err(int ret)
{
        if (ret < 0)
                errno = -ret;
        return ret;
}
```

So it will return a negative number, but errno will still be positive on the error return. Why do you need to set it to negative again ? And also it is supposed to be an unsigned integer:

```go
type Errno uintptr
```

and all general errnos are positive as well:

```go
// Errors
const (
        E2BIG           = Errno(0x7)
        EACCES          = Errno(0xd)
        EADDRINUSE      = Errno(0x62)
        EADDRNOTAVAIL   = Errno(0x63)
        EADV            = Errno(0x44)
        EAFNOSUPPORT    = Errno(0x61)
        EAGAIN          = Errno(0xb)
        EALREADY        = Errno(0x72)
        EBADE           = Errno(0x34)
        EBADF           = Errno(0x9)
        EBADFD          = Errno(0x4d)
        EBADMSG         = Errno(0x4a)
        EBADR           = Errno(0x35)
        EBADRQC         = Errno(0x38)
        EBADSLT         = Errno(0x39)
        EBFONT          = Errno(0x3b)
        EBUSY           = Errno(0x10)
        ECANCELED       = Errno(0x7d)
        ECHILD          = Errno(0xa)
        ECHRNG          = Errno(0x2c)
        ECOMM           = Errno(0x46)
        ECONNABORTED    = Errno(0x67)
        ECONNREFUSED    = Errno(0x6f)
        ECONNRESET      = Errno(0x68)
        EDEADLK         = Errno(0x23)
...
```

Reply:
Also, shouldn't you have a %s to stringify the errno ? Like shown at:

```go
func (e Errno) Error() string {
        if 0 <= int(e) && int(e) < len(errors) {
                s := errors[e]
                if s != "" {
                        return s
                }
        }
        return "errno " + itoa.Itoa(int(e))
}
``` 
Was that your intent here ?

Reply:
There are other places doing the same thing. I'll clarify this first before moving on the review (as I'd like to understand the reasoning of that).

Reply:
Ok you are correct, I didn't mean to have negatives there, that was an inconsistency and mistake :-) 

Also yes, %s should be used since the `error` interface implements stringer. Just missing the %s here.